app/src/androidTest/java/com/github/sdpcoachme/database/CachingStoreTest.kt

```diff
@@ -79,6 +81,105 @@ class CachingStoreTest {
         }
     }
 
+    @Test
+    fun getCoachAverageRatingReturnsAverageRating() {
+        val ratedCoach = COACH_1.copy(
+            ratings = mapOf(
+                NON_COACH_1.email to 0,
+                NON_COACH_2.email to 2
+            )
+        )
+        cachingStore.updateUser(ratedCoach).get(1, SECONDS)
+        val averageRating = cachingStore.getCoachAverageRating(ratedCoach.email)
+            .get(1, SECONDS)
+        assertThat(averageRating, `is`(1))
+    }
+
+    @Test
+    fun getCoachAverageRatingRoundsToNearestInteger() {
+        val ratedCoach1 = COACH_1.copy(
+            ratings = mapOf(
+                NON_COACH_1.email to 0,
+                NON_COACH_2.email to 0,
+                UserInfoSamples.NON_COACH_3.email to 1
+            )
+        )
+        val ratedCoach2 = COACH_2.copy(
+            ratings = mapOf(
+                NON_COACH_1.email to 0,
+                NON_COACH_2.email to 1,
+                UserInfoSamples.NON_COACH_3.email to 1
+            )
+        )
+        val ratedCoach3 = COACH_3.copy(
+            ratings = mapOf(
+                UserInfoSamples.NON_COACH_1.email to 0,
+                UserInfoSamples.NON_COACH_2.email to 5
+            )
+        )
+
+        cachingStore.updateUser(ratedCoach1)
+            .thenCompose { cachingStore.updateUser(ratedCoach2) }
+            .thenCompose { cachingStore.updateUser(ratedCoach3) }
+            .get(1, SECONDS)
+
+
+        val averageRating1 = cachingStore.getCoachAverageRating(ratedCoach1.email)
+            .get(1, SECONDS)
+        assertThat(averageRating1, `is`(0))
+        val averageRating2 = cachingStore.getCoachAverageRating(ratedCoach2.email)
+            .get(1, SECONDS)
+        assertThat(averageRating2, `is`(1))
+        val averageRating3 = cachingStore.getCoachAverageRating(ratedCoach3.email)
+            .get(1, SECONDS)
+        assertThat(averageRating3, `is`(3))
+    }
+
+    @Test
+    fun addRatingToNonCoachUserThrowsException() {
+        cachingStore.setCurrentEmail(NON_COACH_1.email).get(1, SECONDS)
+        cachingStore.updateUser(NON_COACH_2).get(1, SECONDS)
+        try {
+            cachingStore.addRatingToCoach(NON_COACH_2.email, 5).get(1, SECONDS)
```

Comment:
Just as a general comment, I believe that using assertThrows could be written more concise than the assertion done here with the try and catch block. At the end of the day, your code does the same thing, just a little less concise.

Reply:
Indeed you are totally right, however assertThrows is only available in JUNIT 5 and we are using JUNIT 4.

I spent a quiet a lot of time figuring out how to do these tests without assertThrows and this is the cleanest solution I could come up with.

I did not change the JUNIT version to avoid creating problems with the testing suite at the last moment... But this can definitely be done later and a small refactor of those tests will make them way more concise.